modules/backendscheduler/backendscheduler.go

```diff
@@ -0,0 +1,391 @@
+package backendscheduler
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"net/http"
+	"sort"
+	"time"
+
+	"github.com/go-kit/log/level"
+	"github.com/google/uuid"
+	"github.com/grafana/dskit/services"
+	"github.com/grafana/tempo/modules/backendscheduler/work"
+	"github.com/grafana/tempo/modules/overrides"
+	"github.com/grafana/tempo/modules/storage"
+	"github.com/grafana/tempo/pkg/tempopb"
+	"github.com/grafana/tempo/pkg/util/log"
+	"github.com/grafana/tempo/tempodb/blockselector"
+	"github.com/jedib0t/go-pretty/v6/table"
+)
+
+// var tracer = otel.Tracer("modules/backendscheduler")
+
+// BackendScheduler manages scheduling and execution of backend jobs
+type BackendScheduler struct {
+	services.Service
+
+	cfg       Config
+	store     storage.Store
+	overrides overrides.Interface
+
+	work *work.Queue
+}
+
+var _ JobProcessor = (*BackendScheduler)(nil)
+
+// Interface that for future API work to create and manage jobs.
+type JobProcessor interface {
```

Comment:
a lot of these aren't used. i gather this is for future use, but it would help the readability of this change a lot to get it down to only what is necessary to accomplish current goals

Reply:
I've dropped all that isn't used now.  Good call out.